### PR TITLE
Add error tests for File API

### PR DESCRIPTION
**Pull request by xiuqijix**

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-01-05 07:45:46
Finished: 2018-01-05 07:51:02


<h2>Failing Jobs</h2>
<ul>

<li>chrome:dev</li>

<li>safari:10.0</li>

<li>MicrosoftEdge:14.14393</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22414)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/325346775)

File: FileAPI/FileReader/test_securityerrors-manual.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  setup({explicit_done: true});
+  setup({explicit_timeout: true});
+
+  on_event(document.querySelector('#start'), 'click', () => {
+    async_test(t => {
+      let reader = new FileReader();
+      reader.readAsArrayBuffer(fileChooser.files[0]);
+      reader.onloadend = t.step_func_done(event => {
+        assert_equals(event.target.readyState, FileReader.DONE);
+        assert_equals(reader.error.code, 18);
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_securityerrors-manual.html]: Can we test `name` against `"SecurityError"` instead? The `code` property is deprecated.

File: FileAPI/FileReader/test_securityerrors-manual.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  <ol>
+    <li>Select a system sensitive file (e.g. files in /usr/bin, password files,
+        and other native operating system executables) in the file inputbox.</li>
+    <li>Click the 'start' button.</li>
+    </ol>
+</div>
+
+<script>
+
+  let fileChooser = document.querySelector('#fileChooser');
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_securityerrors-manual.html]: nit: could be `const`

File: FileAPI/FileReader/test_securityerrors-manual.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  setup({explicit_done: true});
+  setup({explicit_timeout: true});
+
+  on_event(document.querySelector('#start'), 'click', () => {
+    async_test(t => {
+      let reader = new FileReader();
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_securityerrors-manual.html]: nit: could be `const`

File: FileAPI/FileReader/test_notreadableerrors-manual.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>FileReader NotReadableError Test</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="help" href="http://www.w3.org/TR/FileAPI/#dfn-error-codes">
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_notreadableerrors-manual.html]: Please reference the ED version of the spec: https://w3c.github.io/FileAPI/#dfn-error-codes

File: FileAPI/FileReader/test_securityerrors-manual.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>FileReader SecurityError Test</title>
+<link rel="author" title="Intel" href="http://www.intel.com">
+<link rel="help" href="http://www.w3.org/TR/FileAPI/#dfn-error-codes">
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_securityerrors-manual.html]: Please reference the ED version of the spec: https://w3c.github.io/FileAPI/#dfn-error-codes

File: FileAPI/FileReader/test_notreadableerrors-manual.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    <li>Download the <a href="support/file_test1.txt">file</a>.</li>
+    <li>Select the file in the file inputbox.</li>
+    <li>Delete the file's readable permission.</li>
+    <li>Click the 'start' button.</li>
+  </ol>
+</div>
+
+<script>
+
+  let fileChooser = document.querySelector('#fileChooser');
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_notreadableerrors-manual.html]: nit: could be `const`

File: FileAPI/FileReader/test_notreadableerrors-manual.html [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+  setup({explicit_done: true});
+  setup({explicit_timeout: true});
+
+  on_event(document.querySelector('#start'), 'click', () => {
+    async_test(t => {
+      let reader = new FileReader();
```
Comment (inexorabletash) [inline FileAPI/FileReader/test_notreadableerrors-manual.html]: nit: could be `const`

**Review by inexorabletash — APPROVED** [review]
Thanks for adding these! A few nitpicks...

File: FileAPI/FileReader/test_notreadableerrors-manual.html (hunk 4 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_notreadableerrors-manual.html]: Updated.

File: FileAPI/FileReader/test_notreadableerrors-manual.html (hunk 6 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_notreadableerrors-manual.html]: Updated.

File: FileAPI/FileReader/test_notreadableerrors-manual.html (hunk 7 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_notreadableerrors-manual.html]: Updated.

File: FileAPI/FileReader/test_securityerrors-manual.html (hunk 5 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_securityerrors-manual.html]: Updated.

File: FileAPI/FileReader/test_securityerrors-manual.html (hunk 2 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_securityerrors-manual.html]: Updated.

File: FileAPI/FileReader/test_securityerrors-manual.html (hunk 3 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_securityerrors-manual.html]: Updated.

File: FileAPI/FileReader/test_securityerrors-manual.html (hunk 1 above)
Comment (kaixinjxq) [reply FileAPI/FileReader/test_securityerrors-manual.html]: Updated.

**Comment by kaixinjxq** [comment]
Thanks @inexorabletash, update the nit issues, PTAL.

**Comment by inexorabletash** [comment]
Great, thanks!

**Pull request merged**